src/julia-syntax.scm

```diff
@@ -4131,8 +4131,9 @@ f(x) = yt(x)
                                     `(toplevel-butfirst
                                       ;; wrap in toplevel-butfirst so it gets moved higher along with
                                       ;; closure type definitions
+                                      ,(cadr e)
                                       ,e
-                                      (thunk (lambda () (() () 0 ()) (block (return ,e))))))))
```

Comment:
@JeffBezanson I don't really know why this `thunk` was here. Tests passed without it, but if you happen to remember and I'm missing something, please chime in.

Reply:
Yeah, that's a bit weird. This is 6 years old so maybe something changed around it.

Reply:
It changes the order of evaluation of these statements, see `lift-toplevel`, which relies on the presence of `thunk` to keep the IR ssavalues and struct definitions in the correct order after it chaotically reorders when statements run to follow the toplevel semantic rules